app/scripts/components/analysis/results/use-analysis-params.ts

```diff
@@ -9,40 +9,41 @@ import { polygonUrlDecode, polygonUrlEncode } from '$utils/polygon-url';
 
 // ?start=2015-01-01T00:00:00.000Z&end=2022-01-01T00:00:00.000Z&datasets=no2-monthly|blue-tarp-planetscope&aoi=-9.60205,36.72127|-7.03125,36.87962|-6.85546,39.45316|-6.52587,40.93011|-5.55908,42.11452|-9.38232,42.22851|-8.89892,40.84706|-9.93164,38.85682|-9.47021,38.08268|-8.96484,38.09998||-12.01904,25.95804|-8.65722,25.97779|-8.67919,27.68352|-13.05175,27.68352|-14.89746,25.95804|-12.04101,24.44714
 
-type AnalysisParams =
-  | {
-      date: {
-        start: Date;
-        end: Date;
-      };
-      datasetsLayers: DatasetLayer[];
-      aoi: Feature<MultiPolygon>;
-      errors: null;
-    }
-  | {
-      date: {
-        start: null;
-        end: null;
-      };
-      datasetsLayers: null;
-      aoi: null;
-      errors: null | any[];
-    };
-
-const initialState = {
-  date: {
-    start: null,
-    end: null
-  },
-  datasetsLayers: null,
-  aoi: null,
-  errors: null
+type AnalysisParams = {
+  start: Date;
+  end: Date;
+  datasetsLayers: DatasetLayer[];
+  aoi: Feature<MultiPolygon>;
+  errors: any[] | null;
+};
+
+type AnalysisParamsNull = Omit<Partial<AnalysisParams>, 'errors'> & {
+  errors: any[];
+};
+
+type AnyAnalysisParamsKey = keyof AnalysisParams;
+type AnyAnalysisParamsType = Date | DatasetLayer[] | Feature<MultiPolygon>;
+
+const initialState: AnalysisParamsNull = {
+  start: undefined,
+  end: undefined,
+  datasetsLayers: undefined,
+  aoi: undefined,
+  errors: []
```

Comment:
There are some conflicting types for `errors`.
I'd say we do `undefiend` when there are no errors, and `[]` when there's something. Let's avoid, empty error arrays. thoughts?